### ISSUE TITLE
Is SecureRequestCustomizer needed when the Servlet API is not being used?

### ISSUE THREAD

**Issue by cowwoc** [eclipse/jetty.project#10881]
**Jetty Version**
12.0.3

**Question**
The documentation shows `httpConfig.addCustomizer(new SecureRequestCustomizer())` being used when configuring TLS connections, but the Javadoc description of the class reads: `Customizer that extracts the attribute from an {@link SSLContext} and sets them on the request with {@link Request#setAttribute(String, Object)} according to Servlet Specification Requirements`.

This seems to imply that this class is only required when using the Servlet API. If I am only using the Handler API, can I omit the use of this class?

I would suggest either updating the documentation or Javadoc to clarify this point. Thank you.

**Comment by joakime** [comment]
Yeah, I can see how that documentation is confusing.

The `SecureRequestCustomizer` does update the `Request` attributes from the active endpoint's `SSLEngine` / `SSLContext`, even when using exclusively jetty-core (not servlets).

The Servlet spec is the main beneficiary, but not exclusively.

With the raw `SecureRequestCustomizer` you can find the following new attributes in the Request object.

``` java
request.getAttribute("org.eclipse.jetty.server.cipher");
request.getAttribute("org.eclipse.jetty.server.keySize");
request.getAttribute("org.eclipse.jetty.server.sslSessionId");
request.getAttribute("org.eclipse.jetty.server.peerCertificates");
request.getAttribute("org.eclipse.jetty.server.x509");
request.getAttribute("org.eclipse.jetty.server.sslSession");
request.getAttribute("org.eclipse.jetty.server.sslSessionData");
```

Thing is, if you are using jetty-core, you can access the SSLEngine / SSLContext from the Request object at any time.

``` java
EndPoint endPoint = request.getConnectionMetaData().getConnection().getEndPoint();
if (endPoint instanceof SslEndPoint sslEndPoint)
{
    SslConnection sslConnection = sslEndPoint.getSslConnection();
    SSLEngine sslEngine = sslConnection.getSSLEngine();
    // TODO: get information out of the sslEngine
}
```

**Comment by cowwoc** [comment]
Got it. So I guess I'd suggest to update the Javadoc slightly to clarify that this isn't only for Servlets. Thank you.

**Comment by sbordet** [comment]
Closing, will be addressed in #11597.
